### ISSUE TITLE
Wordclouds Error - ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

### ISSUE THREAD

**Issue by jhyang21** [NLP-Suite/NLP-Suite#1373]
When I try to run the Wordclouds, I get the following error:

Downloading https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.7.0.json: 370kB [00:00, 5.88MB/s]
2024-01-27 17:37:32 INFO: Downloading default packages for language: en (English) ...

No sort order available. Files will be read without sorting.
If you wish to sort the input files in a specific order, you should edit the filename settings using the button 'Setup INPUT/OUTPUT configuration' at the top of the GUI.

2024-01-27 17:37:58 INFO: Checking for updates to resources.json in case models have been updated.  Note: this behavior can be turned off with download_method=None or download_method=DownloadMethod.REUSE_RESOURCES
Downloading https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.7.0.json: 370kB [00:00, 3.37Downloading https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.7.0.json: 370kB [00:00, 3.37MB/s]
2024-01-27 17:37:59 INFO: Loading these models for language: en (English):
========================
| Processor | Package  |
------------------------
| tokenize  | combined |
| mwt       | combined |
========================

2024-01-27 17:37:59 INFO: Using device: cpu
2024-01-27 17:37:59 INFO: Loading: tokenize
2024-01-27 17:38:00 INFO: Loading: mwt
2024-01-27 17:38:00 INFO: Done loading processors!
2024-01-27 17:38:00 INFO: Checking for updates to resources.json in case models have been updated.  Note: this behavior can be turned off with download_method=None or download_method=DownloadMethod.REUSE_RESOURCES
Downloading https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.7.0.json: 370kB [00:00, 3.91MB/s]
Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 715, in urlopen
    httplib_response = self._make_request(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 404, in _make_request
    self._validate_conn(conn)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 1058, in _validate_conn
    conn.connect()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connection.py", line 419, in connect
    self.sock = ssl_wrap_socket(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1073, in _create
    self.do_handshake()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\adapters.py", line 486, in send
    resp = conn.urlopen(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 799, in urlopen
    retries = retries.increment(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\retry.py", line 550, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\packages\six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 715, in urlopen
    httplib_response = self._make_request(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 404, in _make_request
    self._validate_conn(conn)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 1058, in _validate_conn
    conn.connect()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connection.py", line 419, in connect
    self.sock = ssl_wrap_socket(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1073, in _create
    self.do_handshake()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\tkinter\__init__.py", line 1892, in __call__
    return self.func(*args)
  File "wordclouds_main.py", line 134, in <lambda>
    run_script_command=lambda: run(GUI_util.inputFilename.get(),
  File "wordclouds_main.py", line 115, in run
    outputFiles = wordclouds_util.python_wordCloud(inputFilename, inputDir, outputDir, config_filename,  selectedImage,
  File "C:\Users\1025y\Desktop\NLP-Suite-current-stable\src\wordclouds_util.py", line 507, in python_wordCloud
    stannlp = stanza.Pipeline(lang='en', processors=processors)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\pipeline\core.py", line 247, in __init__
    download_models(download_list,
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py", line 531, in download_models
    request_file(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py", line 152, in request_file
    download_file(url, temppath, proxies, raise_for_status)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py", line 114, in download_file
    r = requests.get(url, stream=True, proxies=proxies)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))

![image](https://github.com/NLP-Suite/NLP-Suite/assets/24422126/bc278e58-4ff0-49a6-8eaf-2fe705e64f00)

![image](https://github.com/NLP-Suite/NLP-Suite/assets/24422126/28ff917b-fa9e-4f48-b30e-6a7d0955c575)

**Comment by bianshuyang** [comment]
OK it's a problem jessie seems to encountered sometime ago. What's your stanza version? On mine I type these and obtain 
```
(base) simonbian@SimondeMacBook-Air Test % conda activate NLP
 (NLP) simonbian@SimondeMacBook-Air Test % pip show stanza
**Name: stanza**
**Version: 1.5.1**
Summary: A Python NLP Library for Many Human Languages, by the Stanford NLP Group
Home-page: https://github.com/stanfordnlp/stanza
Author: Stanford Natural Language Processing Group
Author-email: jebolton@stanford.edu
License: Apache License 2.0
Location: /Users/simonbian/anaconda/envs/NLP/lib/python3.10/site-packages
Requires: emoji, numpy, protobuf, requests, torch, tqdm
```

If your version is >=1.6 it might goes to a problem of having a newest model of all languages not yet trained - hence you got a HTTP error.
You need to uninstall stanza's newest release and downgraded to a currently stable version, like 1.5.0 or 1.5.1

**Comment by jhyang21** [comment]
I downgraded it to 1.5.1 from 1.7.0, but the same problem still persists. I also tried it with stanza 1.5.0, but the same problem still persists.

**Comment by Emory-NLP** [comment]
On my machine I have stanza 1.4 and wordcloud 1.9.3 and Pillow 10.2.0 and all works.

**Comment by jhyang21** [comment]
Here are my packages and versions
`(nlp) C:\Users\1025y>pip list
Package                      Version
---------------------------- ------------
absl-py                      2.1.0
annotated-types              0.6.0
asttokens                    2.4.1
astunparse                   1.6.3
autocorrect                  2.3.0
backcall                     0.2.0
bert-extractive-summarizer   0.10.1
blis                         0.7.11
Brotli                       1.0.9
cachetools                   5.3.2
catalogue                    2.0.10
certifi                      2023.11.17
cffi                         1.16.0
chardet                      5.2.0
charset-normalizer           2.0.4
click                        8.1.7
cloudpathlib                 0.16.0
colorama                     0.4.6
confection                   0.1.4
conllu                       4.5.3
contextualSpellCheck         0.4.4
contourpy                    1.1.1
cryptography                 41.0.7
cycler                       0.12.1
cymem                        2.0.8
Cython                       3.0.8
decorator                    5.1.1
editdistance                 0.6.2
emoji                        2.10.0
en-core-web-sm               3.7.1
et-xmlfile                   1.1.0
exceptiongroup               1.2.0
executing                    2.0.1
filelock                     3.13.1
flatbuffers                  23.5.26
fonttools                    4.47.2
fsspec                       2023.12.2
funcy                        2.0
future                       0.18.3
fuzzywuzzy                   0.18.0
gast                         0.4.0
gensim                       4.3.2
geographiclib                1.52
geopy                        2.1.0
gmpy2                        2.1.2
google-auth                  2.26.2
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
googlemaps                   4.4.7
grpcio                       1.60.0
h5py                         3.10.0
huggingface-hub              0.20.2
idna                         3.4
importlib-metadata           7.0.1
importlib-resources          6.1.1
iniconfig                    2.0.0
ipython                      8.12.3
isodate                      0.6.1
jedi                         0.19.1
Jinja2                       3.1.2
joblib                       1.3.2
kaleido                      0.2.1
keras                        2.13.1
kiwisolver                   1.4.5
langcodes                    3.3.0
langdetect                   1.0.7
langid                       1.1.6
Levenshtein                  0.23.0
libclang                     16.0.6
lxml                         4.6.5
Markdown                     3.5.2
MarkupSafe                   2.1.3
matplotlib                   3.3.4
matplotlib-inline            0.1.6
mkl-fft                      1.3.8
mkl-random                   1.2.4
mkl-service                  2.4.0
mlconjug                     3.4.0
mpmath                       1.3.0
murmurhash                   1.0.10
networkx                     3.1
nltk                         3.8.1
numexpr                      2.8.6
numpy                        1.24.4
oauthlib                     3.2.2
openpyxl                     3.0.10
opt-einsum                   3.3.0
packaging                    23.2
pandas                       1.5.2
parso                        0.8.3
pdfminer.six                 20231228
pickleshare                  0.7.5
Pillow                       9.5.0
pip                          23.3.1
plotly                       5.18.0
pluggy                       1.3.0
preshed                      3.0.9
prompt-toolkit               3.0.43
protobuf                     3.20.0
psutil                       5.9.7
pure-eval                    0.2.2
pyasn1                       0.5.1
pyasn1-modules               0.3.0
pycorenlp                    0.3.0
pycparser                    2.21
pydantic                     2.5.3
pydantic_core                2.14.6
pygit2                       1.13.3
Pygments                     2.17.2
pyLDAvis                     2.1.2
pyOpenSSL                    23.2.0
pyparsing                    3.1.1
pyphen                       0.14.0
PySocks                      1.7.1
pyspellchecker               0.8.0
pytest                       7.4.4
python-dateutil              2.8.2
python-docx                  1.1.0
python-Levenshtein           0.23.0
pytz                         2023.3.post1
pywsd                        1.2.5
PyYAML                       6.0.1
rapidfuzz                    3.6.1
rdflib                       7.0.0
regex                        2023.12.25
requests                     2.31.0
requests-oauthlib            1.3.1
rsa                          4.9
safetensors                  0.4.1
scikit-learn                 0.24.2
scipy                        1.10.1
seaborn                      0.13.1
sentence-transformers        2.2.2
sentencepiece                0.1.99
setuptools                   68.2.2
simplekml                    1.3.6
six                          1.16.0
smart-open                   6.4.0
spacy                        3.7.2
spacy-langdetect             0.1.2
spacy-legacy                 3.0.12
spacy-loggers                1.0.5
spacytextblob                4.0.0
SPARQLWrapper                2.0.0
srsly                        2.4.8
stack-data                   0.6.3
stanfordcorenlp              3.9.1.1
stanza                       1.5.0
striprtf                     0.0.26
summarizer                   0.0.7
sympy                        1.12
tenacity                     8.2.3
tensorboard                  2.13.0
tensorboard-data-server      0.7.2
tensorflow                   2.13.0
tensorflow-estimator         2.13.0
tensorflow-intel             2.13.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.4.0
textblob                     0.15.3
textstat                     0.7.3
thinc                        8.2.2
threadpoolctl                3.2.0
tkcolorpicker                2.1.3
tokenizers                   0.15.0
toml                         0.10.2
tomli                        2.0.1
torch                        2.1.2
torchvision                  0.16.2
tqdm                         4.55.2
traitlets                    5.14.1
transformers                 4.36.2
twython                      3.9.1
typer                        0.9.0
typing_extensions            4.9.0
unicode                      2.9
Unidecode                    1.3.8
urllib3                      1.26.18
wasabi                       1.1.2
wcwidth                      0.2.13
weasel                       0.3.4
Werkzeug                     3.0.1
wheel                        0.41.2
win-inet-pton                1.1.0
wn                           0.0.23
wordcloud                    1.8.2.2
wrapt                        1.16.0
zipp                         3.17.0
`

**Comment by bianshuyang** [comment]
If you type to activate a python kernel and initiate appropriate commands as follows, what were the responses? That way I could better understand what problems might be.
From the pip list it appears nothing is wrong, but just in case python behaves unexpectedly.

```
(base) simonbian@SimondeMacBook-Air ~ % conda activate NLP
(NLP) simonbian@SimondeMacBook-Air ~ % python3
Python 3.10.8 (main, Nov 24 2022, 08:08:27) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import stanza
>>> stanza.download('en')
Downloading https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.5.1.json: 366kB [00:00, 20.7MB/s]
2024-01-28 14:42:53 INFO: Downloading default packages for language: en (English) ...
2024-01-28 14:42:55 INFO: File exists: /Users/simonbian/stanza_resources/en/default.zip
2024-01-28 14:43:01 INFO: Finished downloading models and saved to /Users/simonbian/stanza_resources.
>>> stanza.Pipeline(lang='en')
2024-01-28 14:43:07 INFO: Checking for updates to resources.json in case models have been updated.  Note: this behavior can be turned off with download_method=None or download_method=DownloadMethod.REUSE_RESOURCES
Downloading https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.5.1.json: 366kB [00:00, 14.1MB/s]
2024-01-28 14:43:10 INFO: Loading these models for language: en (English):
======================================
| Processor    | Package             |
--------------------------------------
| tokenize     | combined            |
| pos          | combined_charlm     |
| lemma        | combined_nocharlm   |
| constituency | ptb3-revised_charlm |
| depparse     | combined_charlm     |
| sentiment    | sstplus             |
| ner          | ontonotes           |
======================================

2024-01-28 14:43:10 INFO: Using device: cpu
2024-01-28 14:43:10 INFO: Loading: tokenize
2024-01-28 14:43:10 INFO: Loading: pos
2024-01-28 14:43:10 INFO: Loading: lemma
2024-01-28 14:43:10 INFO: Loading: constituency
2024-01-28 14:43:11 INFO: Loading: depparse
2024-01-28 14:43:11 INFO: Loading: sentiment
2024-01-28 14:43:11 INFO: Loading: ner
2024-01-28 14:43:12 INFO: Done loading processors!
<stanza.pipeline.core.Pipeline object at 0x100eeb4f0>
>>> 
```


Also in upgrading and downgrading these packages, did you close the NLP Suite in its entirety? A dependency wouldn't auto-refresh if an instance of a GUI application is in progress.

**Comment by jhyang21** [comment]
I did close the NLP Suite in its entirety. Here is the output:

(base) PS C:\Windows\system32> conda activate NLP
(NLP) PS C:\Windows\system32> python3
python3 : The term 'python3' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ python3
+ ~~~~~~~
    + CategoryInfo          : ObjectNotFound: (python3:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

(NLP) PS C:\Windows\system32> python
Python 3.8.18 (default, Sep 11 2023, 13:39:12) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import stanza
>>> stanza.download('en')
Downloading https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.5.0.json: 216kB [00:00, 4.83MB/s]
2024-01-28 15:13:13 INFO: Downloading default packages for language: en (English) ...
Traceback (most recent call last):
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 715, in urlopen
    httplib_response = self._make_request(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 404, in _make_request
    self._validate_conn(conn)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 1058, in _validate_conn
    conn.connect()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connection.py", line 419, in connect
    self.sock = ssl_wrap_socket(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1073, in _create
    self.do_handshake()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\adapters.py", line 486, in send
    resp = conn.urlopen(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 799, in urlopen
    retries = retries.increment(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\retry.py", line 550, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\packages\six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 715, in urlopen
    httplib_response = self._make_request(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 404, in _make_request
    self._validate_conn(conn)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connectionpool.py", line 1058, in _validate_conn
    conn.connect()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\connection.py", line 419, in connect
    self.sock = ssl_wrap_socket(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 449, in ssl_wrap_socket
    ssl_sock = _ssl_wrap_socket_impl(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\urllib3\util\ssl_.py", line 493, in _ssl_wrap_socket_impl
    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1073, in _create
    self.do_handshake()
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py", line 567, in download
    request_file(
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py", line 152, in request_file
    download_file(url, temppath, proxies, raise_for_status)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py", line 114, in download_file
    r = requests.get(url, stream=True, proxies=proxies)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\requests\adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))
>>>

**Comment by bianshuyang** [comment]
Here's a very convoluted suggestion 1 ( not sure going to work or not)
Please go to:

```File "C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py", line 114, in download_file
r = requests.get(url, stream=True, proxies=proxies)```

As indicated, go to the file

 ```"C:\Users\1025y\anaconda3\envs\NLP\lib\site-packages\stanza\resources\common.py"```

And edit the line 151 and 152 such that you print out the url. Then rerun the program and see how the url is, and whether you could use a browser to download the file.


Another suggestion 2: might try stanza ==1.4.2, or as low as you can get for each one, then see which one might work (by luck!) https://stanfordnlp.github.io/stanza/release_history.html and see if any version after 1.3 and before 1.6 could work

Another suggestion 3: change a wifi network, maybe the ip is banned because of frequent connection to the github page

Another suggestion: close a proxy if available - it is not part of NLP Suite that causes a problem, but simply on how stanza regards its connection to its language models.

**Comment by jhyang21** [comment]
Suggestion 1: url shows https://raw.githubusercontent.com/stanfordnlp/stanza-resources/main/resources_1.4.1.json, but I'm not sure how to download this and implement.

Suggestion 2: did not work with 1.3.0 to 1.4.2 

Suggestion 3: did not work

Suggestion 4: not sure how to close a proxy

I did upgrade numpy in nlp environment before because I got a incompatibility warning from something else. Could you specify what version of numpy you are running with?

**Comment by bianshuyang** [comment]
Version: 1.24.1
And I am out of solutions for the moment. Here's all what I have
[my_pips.txt](https://github.com/NLP-Suite/NLP-Suite/files/14078598/my_pips.txt)

**Comment by jhyang21** [comment]
I tried downloading the resources_1.4.1.json and putting it in the same directory as common.py, but that didn't fix the problem.
I tried turning off the firewall, but that didn't fix the problem.
I tried `pip install -r requirements.txt` in the src directory, but that didn't fix the problem.

I remember running the same command in 4.5.9 version of NLP-suite and it worked for me, but I'm not sure why it doesn't anymore.
